### ISSUE TITLE
Better error message for RectangularDomain.sorted()

### ISSUE THREAD

**Issue by jabraham17** [chapel-lang/chapel#25109]
Currently we have the `.sorted` method on domains. It is only intended to work on associative domains, although its documentation does not indicate this. However, should a user try to use `.sorted` on a rectangular domain they get a confusing error from an internal module.

```chapel
var d = {1..10};
var s = d.sorted(); // error: unresolved call 'unmanaged domain(1,int(64),one).dsiSorted(DefaultComparator)'
writeln(s);
```

This contrasts with other non rectangular domain methods like `bulkAdd`

```chapel
var d = {1..10};
d.bulkAdd([1,2]); // error: incompatible argument(s) or this domain type does not support 'bulkAdd'
```

We should have a better error message for `.sorted`, possibly implemented as an error overload in `ChapelDomain.chpl`

**Comment by Aenk-Singhal** [comment]
I am a newbie so dont know much saw this was a good first issue would like to try it.

**Comment by jabraham17** [comment]
Great! As mentioned in the original post, see `proc bulkAdd` in `modules/internal/ChapelDomain.chpl` for an example of other error overloads for domain methods. Then to make sure that your changes are tested, add a test of the error message in `test/domains/compilerErrors/`, `test/domains/compilerErrors/notsupportedBulkAdd-assoc.chpl` is probably a good example.

I recommend taking a quick look at [these docs](https://chapel-lang.org/docs/main/developer/bestPractices/ContributorInfo.html) for how to contribute to the Chapel repo. In summary:
1. Create a new fork of the chapel repo and clone your new fork
2. Create a development branch for this feature
3. Add your changes
4. Test your changes
  - Assuming you have installed [the prerequisites](https://chapel-lang.org/docs/main/usingchapel/prereqs.html), run the following commands
    ```bash
    # from the chapel repo locally
    source util/setchplenv.bash
    make -j
    make test-venv
    start_test path/to/the/test.chpl
    ```
  - You can also build the docs, you can open `build/doc/html/index.html` in your browser
    ```bash
     make chpldoc docs -j
    ```
5. Create a pull request with your branch and find someone to review it. Feel free to ping me on it

Feel free to reach out if you have questions!

**Comment by yashathwani** [comment]
Would like to work on this issue could you assign it to me !!

**Comment by jabraham17** [comment]
Hi @yashathwani, I think @Aenk-Singhal is currently working on this. However, if they don't respond to this in about a week you should consider this issue fair game and go ahead and open a PR for it.

**Comment by jabraham17** [comment]
@yashathwani  and @Aenk-Singhal, just noting that the goal for this issue is to get it resolved for the 2.2 release, which is happening in September. This means that the implementation for this PR has to be done and merged by feature freeze on September 5th.

**Comment by jabraham17** [comment]
Noting that as feature freeze is 2 days away, I have gone ahead and opened https://github.com/chapel-lang/chapel/pull/25863 which will implement the better error message and resolve this issue
